consult.el

```diff
@@ -574,12 +594,24 @@ We use invalid characters outside the Unicode range.")
 
 Turn ARG into a list, and for each element either:
 - split it if it a string.
-- eval it if it is an expression."
-  (mapcan (lambda (x)
-            (if (stringp x)
-                (split-string-and-unquote x)
-              (ensure-list (eval x 'lexical))))
-          (ensure-list arg)))
+- eval it if it is an expression.
+
+If `consult--search-path-list' is set and not equal to '(\".\"),
+splice its paths in at the end of the arg list."
+  (let ((args (copy-sequence (mapcan (lambda (x)
+				       (if (stringp x)
+					   (split-string-and-unquote x)
+					 (ensure-list (eval x 'lexical))))
+				     (ensure-list arg)))))
+    (if (and consult--search-path-list
+	     (not (equal consult--search-path-list '("."))))
+	(if (member "." args)
+	    (progn (unless (and consult--search-path-list
+				(file-directory-p (car consult--search-path-list)))
+		     (setq args (delete "-r" args)))
+		   (flatten-list (consult--replace-in-list "." consult--search-path-list args)))
+	  (nconc args consult--search-path-list))
+      args)))
```

Comment:
`consult--build-args` is not the correct place to add this code, since this function should be generic for all commands. It should rather go to the different command builders, `consult--grep-make-builder` etc. These builders must be reworked anyway to avoid the global variables.